pkg/kubelet/kuberuntime/kuberuntime_sandbox.go

```diff
@@ -169,6 +169,15 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxLinuxConfig(pod *v1.Pod) (
 		}
 		lc.SecurityContext.NamespaceOptions = namespacesForPod(pod)
 
+		if m.runtimeConfig != nil && m.runtimeConfig.IsUserNamespaceSupported() && !userNamespaceDefinedForPod(pod) {
```

Comment:
Should we raise an error when `userNamespaceForPod` is `NamespaceMode_NODE_WIDE_REMAPPED` and the runtime doesn't support it?

Reply:
I think we should raise an error:
https://github.com/vikaschoudhary16/community/blob/1cfe64b48107575c98cef4cd321e435482c60293/contributors/design-proposals/node/node-usernamespace-remapping.md#pod-admission-at-kubelet

> A pod admit handler will be introduced which will fail the pod admission at Kubelet [...]
> If pod spec has hostuserNamespace: false AND usernamespace remapping is NOT enabled at container runtime

But I don't know if it is the right place for this check. Shouldn't it be done before?